mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryIntegrationResources.java

```diff
@@ -200,6 +200,41 @@ private Resources createResources(EventBus eventBus, GroupMembershipResolver gro
         return new Resources(manager, storeRightManager, new InMemoryMessageId.Factory(), currentQuotaManager, quotaRootResolver, maxQuotaManager);
     }
 
+    public Resources createResources(SessionProvider sessionProvider, InMemoryMailboxSessionMapperFactory mailboxSessionMapperFactory, Set<PreDeletionHook> preDeletionHooks) {
```

Comment:
I'd like to not have duplication between the createResources freaks method and write the ugly job only once...

Reply:
But for create `DeletedMessageVaultHook`, I need `sessionProvider`, `mapperFactory`. And for `createResources` we can not argument it. It is chicken-eggs problem. So I need to create a new method for this.

Reply:
Ahhh. I see your points.

Let's go with this, I think this class needs some serious refactoring anyway.